process/transaction/interceptedTransaction.go

```diff
@@ -191,3 +192,30 @@ func (inTx *InterceptedTransaction) Transaction() *transaction.Transaction {
 func (inTx *InterceptedTransaction) Hash() []byte {
 	return inTx.hash
 }
+
+// GetSenderShardId returns the transaction sender shard id
+func (inTx *InterceptedTransaction) GetSenderShardId() uint32 {
+	return inTx.sndShard
+}
+
+// GetNonce return the transaction nonce
+func (inTx *InterceptedTransaction) GetNonce() uint64 {
+	return inTx.tx.Nonce
+}
+
+// GetSenderAddress return the transaction sender address
+func (inTx *InterceptedTransaction) GetSenderAddress() state.AddressContainer {
+	return inTx.sndAddr
+}
+
+// GetTotalValue returns the maximum cost of transaction
```

Comment:
maybe write the formula in the comment: `totalValue = txValue + gasPrice*gasLimit`

Reply:
Done.